kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import {useState} from "react";
+import * as _ from "lodash";
+import "./ExpressionContainer.css"
+import {useBoxedExpressionEditorI18n} from "../../i18n";
+import {
+  Dropdown,
+  DropdownItem,
+  KebabToggle,
+  Popover,
+  SimpleList,
+  SimpleListItem,
+  SimpleListItemProps
+} from "@patternfly/react-core";
+
+export interface ExpressionContainerProps {
+  /** The name of the expression */
+  name: string,
+  /** The type of the expression */
+  type?: string,
+  /** Selected expression is already present */
+  selectedExpression?: string
+}
+
+const ExpressionContainer: (props: ExpressionContainerProps) => JSX.Element = (props: ExpressionContainerProps) => {
+  const {i18n} = useBoxedExpressionEditorI18n();
+
+  const [logicTypeIsPresent, isLogicTypeSet] = useState(!_.isEmpty(props.selectedExpression));
+  const [actionDropdownIsOpen, isActionDropdownOpen] = useState(false);
+
+  const hideSelectorMenuPopover = () => {
```

Comment:
When I opened the Guided Tour PR I had a lot of functions like this too ☺️. However the `kogito-tooling` team advised me that every function inside of a React component must be created by using the `useCallback`, so I believe we should follow that guide-line (still, this guide-line can be a bit [controversial](https://dmitripavlutin.com/dont-overuse-react-usecallback), I believe we should apply on some functions in this component) ;-)